tracing/src/main/java/com/palantir/remoting3/tracing/Tracer.java

```diff
@@ -48,8 +52,8 @@ private Tracer() {}
         return trace;
     });
 
-    // Thread-safe Map implementation
-    private static final Map<String, SpanObserver> observers = Maps.newConcurrentMap();
+    private static final Map<String, SpanObserver> observers = new HashMap<>();
```

Comment:
`@GuardedBy("this")`

Reply:
that annotation is currently not on the classpath :( I added a comment to the effect given that GuardedBy isn't really read by anything.

Reply:
That's sad as error-prone will do some validations when annotated: https://github.com/google/error-prone/blob/master/docs/bugpattern/GuardedBy.md